LEAF_Nexus/sources/Employee.php

```diff
@@ -75,6 +77,460 @@ public static function cleanWildcards($input)
+     */
+    private function userNameUidList(array $employee_list): array
+    {
+        $return_value = [];
+
+        foreach ($employee_list as $employee) {
+            $return_value[$employee['userName']] = $employee['empUID'];
+        }
+
+        return $return_value;
+    }
+
+    /**
+     * @param Db $db
+     *
+     * @return array
+     *
+     * Created at: 6/9/2023, 2:30:36 PM (America/New_York)
+     */
+    private function getAllEmployees(Db $db): array
+    {
+        $vars = array();
+        $sql = 'SELECT userName
+                FROM employee';
+
+        $return_value = $db->prepared_query($sql, $vars);
+
+        return $return_value;
+    }
+
+    /**
+     * @param array $user
+     *
+     * @return bool
+     *
+     * Created at: 6/9/2023, 2:30:52 PM (America/New_York)
+     */
+    private function isActiveNationally(array $user): bool
+    {
+        $weekOld = $user[0]['lastUpdated'] + 604800;
+
+        if (time() < $weekOld) {
+            $return_value = true;
+        } else {
+            $return_value = false;
+        }
+
+        return $return_value;
+    }
+
+    /**
+     * @param array $user_names
+     * @param Db $db
+     *
+     * @return array
+     *
+     * Created at: 6/9/2023, 2:31:07 PM (America/New_York)
+     */
+    private function getEmployeeByUserName(array $user_names, Db $db): array
+    {
+        $sql = "SELECT empUID, userName, lastName, firstName, middleName,
+                    phoneticLastName, phoneticFirstName, domain, deleted,
+                    lastUpdated
+                FROM employee
+                WHERE userName IN (" . implode(",", array_fill(1, count($user_names), '?')) . ")";
+        $return_value = $db->prepared_query($sql, $user_names);
+
+        return $return_value;
+    }
+
+    /**
+     * @param array $empUID
+     * @param Db $db
+     *
+     * @return array
+     *
+     * Created at: 6/9/2023, 2:31:14 PM (America/New_York)
+     */
+    private function getEmployeeDataByEmpUID(array $empUID, Db $db): array
+    {
+        $vars = array(':PHONEIID' => 5,
+                ':EMAILIID' => 6,
+                ':LOCATIONIID' => 8,
+                ':ADTITLEIID' => 23
+            );
+        $sql = "SELECT empUID, employee.userName, indicatorID, data, author, timestamp
+                FROM employee_data
+                LEFT JOIN employee USING (empUID)
+                WHERE empUID IN ('" . implode("','", array_values($empUID)) . "')
+                AND indicatorID IN (:PHONEIID,:EMAILIID,:LOCATIONIID,:ADTITLEIID)";
+        $return_value = $db->prepared_query($sql, $vars);
+
+        return $return_value;
+    }
+
+    /**
+     * @param string $user_name
+     * @param array $national_user
+     * @param Db $db
+     *
+     * @return void
+     *
+     * Created at: 6/9/2023, 2:31:54 PM (America/New_York)
+     */
+    private function updateEmployeeByUserName(string $user_name, array $national_user, Db $db): void
```

Comment:
But here as well.  I don't think something like this should be "fire and forget".  In this case, it's common for the return to include the results of a query that demonstrates all the data was correctly updated.

Reply:
I am returning the data that was updated, if you look at where this method is called you will see the next line gets the data to be returned. I double checked to be sure, an update sql statement doesn't return anything so unless I query the date here instead of after the call like I am now there is nothing really to send back.

Reply:
Ok.  Then just a "success" json return